test/core.test.js

```diff
@@ -15,28 +15,221 @@ describe('ravelin.core', function() {
+        "exp": {
+          "device": /rjs-[a-z0-9-]{30,}/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "session": "sess-cook"
+        },
+        "cfg": {
+          "id": "opt-id"
+        },
+        "exp": {
+          "device": /opt-id/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "session": "sess-dev-cook:sess-cook"
+        },
+        "cfg": {},
+        "exp": {
+          "device": /sess-dev-cook/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "session": "sess-dev-cook:sess-cook"
+        },
+        "cfg": {
+          "id": "opt-id"
+        },
+        "exp": {
+          "device": /opt-id/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "device": "dev-cook"
+        },
+        "cfg": {},
+        "exp": {
+          "device": /dev-cook/,
+          "session":/[a-z0-9-]{30,}/
+        }
+      },
+      {
+        "cookies": {
+          "device": "dev-cook"
+        },
+        "cfg": {
+          "id": "opt-id"
+        },
+        "exp": {
+          "device": /opt-id/,
+          "session":/[a-z0-9-]{30,}/
+        }
+      },
+      {
+        "cookies": {
+          "device": "dev-cook",
+          "session": "sess-cook"
+        },
+        "cfg": {},
+        "exp": {
+          "device": /dev-cook/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "device": "dev-cook",
+          "session": "sess-cook"
+        },
+        "cfg": {
+          "id": "opt-id"
+        },
+        "exp": {
+          "device": /opt-id/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "device": "dev-cook",
+          "session": "sess-dev-cook:sess-cook"
+        },
+        "cfg": {},
+        "exp": {
+          "device": /sess-dev-cook/,
+          "session":/sess-cook/
+        }
+      },
+      {
+        "cookies": {
+          "device": "dev-cook",
+          "session": "sess-dev-cook:sess-cook"
+        },
+        "cfg": {
+          "id": "opt-id"
+        },
+        "exp": {
+          "device": /opt-id/,
+          "session":/sess-cook/
+        }
+      }
+    ];
+    tc.forEach(function(t) {
+      it('passes ' + JSON.stringify(t), function() {
+        var cfg = isolate(t.cfg || {});
+        if (t.cookies) {
+          if (t.cookies.device) document.cookie = cfg.cookie + '=' + t.cookies.device;
+          if (t.cookies.session) document.cookie = cfg.sessionCookie + '=' + t.cookies.session;
+        }
+        var r = new Ravelin(cfg);
+        return r.core.ids().then(function(ids) {
+          expect(ids.device).to.match(t.exp.device);
+          expect(ids.session).to.match(t.exp.session);
+          expect(r.core.cookies.get(cfg.sessionCookie)).to.be(ids.device + ':' + ids.session);
+          if (!(cfg.cookieExpiryDays <= 0)) {
```

Comment:
This might be easier to read:

```suggestion
          if (cfg.cookieExpiryDays > 0) {
```

Reply:
This block should run when cfg.cookieExpiryDays is undefined.

Reply:
Actually... there aren't any cases that have a cookieExpiryDays <= 0, so I should just be checking the cookie every time.

Reply:
Removed in https://github.com/unravelin/ravelinjs/pull/54/commits/ac6bbfffecb0403af0f0c1c817ebd6f3f9aa4998.